### PR TITLE
Start of a 2.0 project loader

### PR DESCRIPTION
**Pull request by tmickel**
Pretty exciting, I think. Gets most of the way through #96, minus importing actual files. It loads a bunch of projects from the website (of course, none of them behave as expected).

Demo: https://www.dropbox.com/s/gek6nndums3x6fr/sb2-demo.mov?dl=0

maybe best approached commit-by-commit. I tried to structure them as a logical progression.

Things to look at in particular:
- Blocks.prototype.toXML - using this to generate a Blockly workspace for the loaded blocks.
- Editing targets (the currently selected sprite, for example)
- The entire file "src/import/sb2import.js" including the spec-map approach. It's kind of boilerplate madness™ so maybe there are other approaches. Thoughts about requirements: shouldn't require loading Blockly (so a programmatic solution using `Blockly.Blocks` or toolbox XML wouldn't work), should work offline. If it feels like too much boilerplate, maybe we should consider not changing 2.0 opcodes and trying to use positional args somehow. However, the structure of how it is now does feel like it's pretty nice for some things. And maybe it's not so bad to update the table as we change what 2.0 blocks map to.

cc: @cwillisf

### CODE REVIEW

File: src/engine/blocks.js [hunk 1]
```diff
@@ -120,63 +120,57 @@ Blocks.prototype.getInputs = function (id) {
...
-        if (typeof e !== 'object') return;
-        if (typeof e.blockId !== 'string') return;
-
-        // UI event: clicked scripts toggle in the runtime.
-        if (e.element === 'stackclick') {
-            if (opt_runtime) {
-                opt_runtime.toggleScript(e.blockId);
-            }
-            return;
+Blocks.prototype.blocklyListen = function (e, isFlyout, opt_runtime) {
```
Comment (tmickel) [inline src/engine/blocks.js:128]: (note nothing changed here, other than no longer generating a new function)

**Comment by TheBrokenRail** [comment]
Awesome! Variables don't work though. when a variable is in a blocks it changes the input it's in to no input dark square. When notr in block it turns to unknown block

**Comment by tmickel** [comment]
@TheBrokenRail Hmm, yeah, thanks. I think to avoid ballooning this PR I'd like to address that separately.

File: src/index.js [hunk 2]
```diff
@@ -101,6 +101,65 @@ VirtualMachine.prototype.postIOData = function (device, data) {
...
+/**
+ * Set an editing target. An editor UI can use this function to switch
+ * between editing different targets, sprites, etc.
+ * After switching the editing target, the VM may emit updates
+ * to the list of targets and any attached workspace blocks
+ * (see `emitTargetsUpdate` and `emitWorkspaceUpdate`).
+ * @param {string} targetId Id of target to set as editing.
+ */
+VirtualMachine.prototype.setEditingTarget = function (targetId) {
+    if (targetId == this.editingTarget.id) { // No change.
```
Comment (thisandagain) [inline src/index.js]: Total nit: not sure about the structure of the comments here. 

``` js
// No change
if (foo === bar) return;

// Do things
// blah
```

File: playground/playground.js [hunk 3]
```diff
@@ -71,6 +90,38 @@ window.onload = function() {
         updateBlockExplorer(data.blocks);
     });
 
+    vm.on('workspaceUpdate', function (data) {
```
Comment (thisandagain) [inline playground/playground.js:101]: Might be good to add some documentation for each vm event handler.

**Comment by TheBrokenRail** [comment]
I think it's works well enough to be merged.

**Comment by thisandagain** [comment]
This is really exceptional work. Thanks @tmickel . I think this is good to land, though I think it might be good to add issues to track 1.4 support as well as unit testing for the parser / loader.

**Comment by dekrain** [comment]
I'm ready to update my server :smile:

**Comment by tmickel** [comment]
Thanks @thisandagain ! Sounds good, and I'm sure there are bugs here that we'll discover, too.

**Pull request merged**